ethers-providers/src/rpc/transports/ws/manager.rs

```diff
@@ -234,6 +240,70 @@ impl RequestManager {
         ))
     }
 
+    #[cfg(not(target_arch = "wasm32"))]
```

Comment:
can we DRY these? this and `reconnect()` are largely duplicated code. take the part that needs to be conditionally compiled and move it into a smaller function instead of duplicating the entire function

Reply:
should be better now